### ISSUE TITLE
Create GitHub Action To Lint Cargo.toml files with dprint

### ISSUE THREAD

**Issue by metaclips** [build-trust/ockam#4637]
Currently, our Cargo.toml files don't get linted on CI which allows poorly formatted toml file, we should lint our toml file with [dprint](https://github.com/marketplace/actions/dprint-check-action) ensuring CI fail if Cargo.toml is not properly formated

**Comment by itsajay1029** [comment]
Can you assign this issue to me @metaclips ? I'd love to work on it

**Comment by metaclips** [comment]
Yeah, sure.

**Comment by itsajay1029** [comment]
Can you explain me the issue a little ? What needs to be done ?? @metaclips

**Comment by metaclips** [comment]
@itsajay1029 we need to add Dprint to check our codebase if there are Cargo.toml files that are not properly formatted.

To install Dprint on GitHub action, we can call
```bash
curl -fsSL https://dprint.dev/install.sh | sh
```

We also need to specify a json file (dprint.json) https://dprint.dev/plugins/toml/ with set of formatting rules in the root directory, the json specified below should be sufficient
```json
{
    "includes": [
        "**/Cargo.toml"
    ],
    "plugins": [
        "https://plugins.dprint.dev/toml-0.5.4.wasm"
    ]
}
```

We can then call `dprint check` to check if all files are properly formatted

We should add the dprint CI in the `all workflow` https://github.com/build-trust/ockam/blob/develop/.github/workflows/all.yml and name it `lint_cargo_toml`.

Also dprint has an awesome doc here https://dprint.dev/overview/. Please feel free to ask more questions here if any.

**Comment by itsajay1029** [comment]
I have a Windows system... Will I use npm for installing dprint ? @metaclips

**Comment by metaclips** [comment]
> I have a Windows system... Will I use npm for installing dprint ? @metaclips

To test dprint locally on your system on windows, you can install using powershell https://dprint.dev/install/
```bash
iwr https://dprint.dev/install.ps1 -useb | iex
```
Or you can install with Rust Cargo using
```bash
cargo install --locked dprint
```

**Comment by itsajay1029** [comment]
Is it fine if I add this under the jobs section of `all-workflow` @metaclips  ?

```
lint-cargo-toml:
    name: lint-cargo-toml
    runs-on: ubuntu-20.04
    steps:
    - name: Checkout code
      uses: actions/checkout@v2
    - name: Install dprint
      run: sudo snap install dprint
    - name: Check formatting
      run: |
        dprint check 
    - name: Apply formatting
      run: |
        dprint fmt 
```

**Comment by metaclips** [comment]
> Is it fine if I add this under the jobs section of `all-workflow` @metaclips ?
> 
> ```
> lint-cargo-toml:
>     name: lint-cargo-toml
>     runs-on: ubuntu-20.04
>     steps:
>     - name: Checkout code
>       uses: actions/checkout@v2
>     - name: Install dprint
>       run: sudo snap install dprint
>     - name: Check formatting
>       run: |
>         dprint check 
>     - name: Apply formatting
>       run: |
>         dprint fmt 
> ```

Yeah @itsajay1029. Let's create a follow up PR we can review and make changes needed.

**Comment by itsajay1029** [comment]
Okay, I have created a pull request @metaclips  : https://github.com/build-trust/ockam/pull/4723

**Comment by metaclips** [comment]
fixed in #4723 thanks @itsajay1029
